internal/app/app.go

```diff
@@ -1411,33 +1339,6 @@ func (app *App) performSwitchover(clusterState map[string]*NodeState, activeNode
 	return nil
 }
 
-func (app *App) getCurrentMaster(clusterState map[string]*NodeState) (string, error) {
-	var master string
-	err := app.dcs.Get(pathMasterNode, &master)
-	if err != nil && err != dcs.ErrNotFound {
-		return "", fmt.Errorf("failed to get current master from dcs: %s", err)
-	}
-	if master != "" {
-		return master, nil
-	}
-	return app.ensureCurrentMaster(clusterState)
-}
-
-func (app *App) ensureCurrentMaster(clusterState map[string]*NodeState) (string, error) {
```

Comment:
Inside this method app.getMasterHost is called, so we cannot just move it inside app_dcs.

You should extract GetMaster(...) and SetMaster(...) methods, which only operates with dcs, and called inside getCurrentMaster and ensureCurrentMaster.